planner/core/rule_predicate_push_down.go

```diff
@@ -99,6 +99,7 @@ func (p *LogicalUnionScan) PredicatePushDown(predicates []expression.Expression)
 
 // PredicatePushDown implements LogicalPlan PredicatePushDown interface.
 func (ds *DataSource) PredicatePushDown(predicates []expression.Expression) ([]expression.Expression, LogicalPlan) {
+	predicates = expression.PropagateConstant(ds.ctx, predicates)
```

Comment:
Maybe we would do `expression.PropagateConstant` many times (much more than necessary).